plone/app/dexterity/browser/types.py

```diff
@@ -76,41 +76,46 @@ def handleClone(self, action):
             self.status = _(u'Please select a single type to clone.')
         elif len(selected) == 1:
             id = selected[0][0]
```

Comment:
I didn't review the changes myself, I did a first draft and then pushed to check it with jenkins, there's a few things that need to be fixed indeed.

About this id, we are shadowing it so much that I even wrote a flake8 plugin for that, although so far we don't apply it as it will give way too much errors.